api/src/main/java/marquez/api/models/ListResult.java

```diff
@@ -0,0 +1,31 @@
+package marquez.api.models;
+
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.Collections;
+import java.util.Map;
+import lombok.NonNull;
+
+public class ListResult<T> {
```

Comment:
Maybe `ResultsPage<T>`? Maybe also include the `offset` in this structure?

Reply:
I'd prefer the use of `ResultsPage<T>` as well, as it makes it clear the class is used for offset-based pagination. In addition to returning the `offset` and `totalCount`, I'd also add the `limit` to the response.

For clarification, `totalCount` is equal to the total number of _current_ entries in the `datasets`, `jobs`, etc tables?

Reply:
Sounds good on renaming the class.

Aren't `offset` and `limit` already known values? Those values would be reflecting input parameters back at the user; I'm not sure I see the value in that.

The `totalCount` value that I'm adding is so a client can know how many pages of data there are total when requesting any page.